frontends/verilog/verilog_parser.y

```diff
@@ -2530,6 +2531,7 @@ gen_stmt:
 		ast_stack.push_back(node);
 		ast_stack.back()->children.push_back($3);
 	} gen_stmt_block opt_gen_else {
+		SET_AST_NODE_LOC(ast_stack.back(), @1, @7);
```

Comment:
I still do not know if this should be `(@1, @6)` or `(@1, @7)`.  Should the `opt_gen_else` be included in the `if` location?  Does the `opt_gen_else` have its own location assigned?  It looks like it doesn't.  I am not sure exactly how this generate stuff is parsed and represented as an AST, but if I have:
```
<<EOT
if (condition1)
...
else
...
EOT
```

Doesn't it make more sense for the `AST_GENIF` to be at `1.1-2.X` and the `opt_gen_else` node (if it exists?) be at `3.1-4.Y` rather than the `AST_GENIF` to be at `1.1-4.Y`?

Reply:
Looking at the `opt_gen_else` clause there isn't really an obvious else node, so I think it makes sense to associate it with the `if`.

Reply:
OK, that's fine, then.  BTW, do you have an example `-dump-ast1` output from a relevant input file?

Reply:
```verilog
module top(input a, output q);
	parameter MODE = 1'b0;
	generate
	if (MODE)
		assign q = !a;
	else
		assign q = a;
	endgenerate
endmodule
```

gives
```
    AST_MODULE <genif.v:1.1-9.10> [0x5628d920e900] str='\top'
      AST_WIRE <genif.v:1.18-1.19> [0x5628d920dde0] str='\a' input port=1
      AST_WIRE <genif.v:1.28-1.29> [0x5628d920df20] str='\q' output port=2
      AST_PARAMETER <genif.v:0.0-0.0> [0x5628d920e680] str='\MODE'
        AST_CONSTANT <genif.v:0.0-0.0> [0x5628d920e2e0] bits='0'(1) range=[0:0]
      AST_GENIF <genif.v:4.2-7.16> [0x5628d920e1a0]
        AST_IDENTIFIER <genif.v:4.6-4.10> [0x5628d920e060] str='\MODE'
        AST_GENBLOCK <genif.v:5.3-5.17> [0x5628d920e7c0]
          AST_ASSIGN <genif.v:5.10-5.16> [0x5628d920eb80]
            AST_IDENTIFIER <genif.v:5.10-5.11> [0x5628d9209d40] str='\q'
            AST_LOGIC_NOT <genif.v:5.14-5.16> [0x5628d920ede0]
              AST_IDENTIFIER <genif.v:5.15-5.16> [0x5628d920ef20] str='\a'
        AST_GENBLOCK <genif.v:7.3-7.16> [0x5628d92091d0]
          AST_ASSIGN <genif.v:7.10-7.15> [0x5628d9218ab0]
            AST_IDENTIFIER <genif.v:7.10-7.11> [0x5628d9207de0] str='\q'
            AST_IDENTIFIER <genif.v:7.14-7.15> [0x5628d9209860] str='\a'
```

Reply:
OK, cool, thanks.

Reply:
It doesn't seem related to this PR in particular though, as the same behaviour occurs with regular, non-generate if statements too:
```verilog
module top(input a, output reg q);
	always @*
		if (1'b1)
			q = !a;
endmodule
```
gives
```
    AST_MODULE <if.v:1.1-5.10> [0x55aa0194cf20] str='\top'
      AST_WIRE <if.v:1.18-1.19> [0x55aa0194c2e0] str='\a' input port=1
      AST_WIRE <if.v:1.32-1.33> [0x55aa0194c680] str='\q' output reg port=2
      AST_ALWAYS <if.v:2.2-4.11> [0x55aa0194c7c0]
        AST_BLOCK <if.v:2.11-4.11> [0x55aa0194c900]
          AST_CASE <if.v:3.3-4.11> [0x55aa0194cb80]
            AST_REDUCE_BOOL <if.v:0.0-0.0> [0x55aa01945de0]
              AST_CONSTANT <if.v:0.0-0.0> [0x55aa0194ca40] bits='1'(1) range=[0:0] int=1
            AST_COND <if.v:3.7-3.11> [0x55aa0194c1a0]
              AST_CONSTANT <if.v:0.0-0.0> [0x55aa019471d0] bits='1'(1) range=[0:0] int=1
              AST_BLOCK <if.v:4.4-4.11> [0x55aa0194cde0]
                AST_ASSIGN_EQ <if.v:4.4-4.10> [0x55aa0194bf20]
                  AST_IDENTIFIER <if.v:4.4-4.5> [0x55aa01947860] str='\q'
                  AST_LOGIC_NOT <if.v:4.8-4.10> [0x55aa0194bde0]
                    AST_IDENTIFIER <if.v:4.9-4.10> [0x55aa01947d40] str='\a'
```

Reply:
Hmm, indeed, I see.  Maybe this is that annoying longstanding bug with `TOK_ID` that `verilog_parser.y` has a FIXME about...